File: 35.py
# mid split search
class Solution:
    def searchInsert(self, nums, target):
        if len(nums) <= 0:
            return 0
        min = 0
        max = len(nums) - 1
        mid = (min + max)//2
        while True:
            if target <= nums[0]:
                return min
            if target > nums[max]:
                return max + 1
            if target ==nums[max]:
                return max
            mid = (min + max)//2
            if target > nums[mid]:
                min = mid + 1
            elif target < nums[mid]:
                max = mid - 1
            else:
                return mid

s = Solution()
print(s.searchInsert([1], 1))

File: 35.ts
namespace T35 {
  function searchInsert(nums: number[], target: number): number {
    if (nums.length === 0) {
      return 0;
    }
    let head = 0,
      tail = nums.length - 1,
      mid;

    while (head < tail) {
      mid = Math.floor((head + tail) / 2);
      if (nums[mid] === target) {
        return mid;
      } else if (nums[mid] > target) {
        tail = mid - 1;
      } else {
        head = mid + 1;
      }
    }

    if (nums[head] >= target) {
      return head;
    } else {
      return head + 1;
    }
  }
}
